### PR TITLE
feat(Accordion): close all `Accordions` in `Accordion.Group`

### PR DESCRIPTION
**Pull request by joakbjerk**
Add functionality to close all `Accordions` in a `Accordion.Group`

Based on https://github.com/dnbexperience/eufemia/pull/3666 🙏

### CODE REVIEW

**Review by langz — APPROVED** [review]
Nice, code looks good, I'll try to quickly demo/test it when the deploy preview is ready 🙏

**Comment by langz** [comment]
After demoing/testing the example provided, I noticed an anomaly 👽 

Close All functionality seems to work perfect ⭐ 
When pressing/closing one of the accordions, all of the other accordions close as well, is that normal/expected? 
Personally, I would expect only the one I'd click to close?

https://github.com/dnbexperience/eufemia/assets/1359205/ff9314fe-a550-4ab7-9d6e-a6194f34fc0f


I have not given this much thought(and not looked into the code of the example, etc), and it probably doesn't matter for the example which demos the "close all functionality", but I just thought of sharing this.

**Comment by joakbjerk** [comment]
> After demoing/testing the example provided, I noticed an anomaly 👽 
> 
> Close All functionality seems to work perfect ⭐ 
> When pressing/closing one of the accordions, all of the other accordions close as well, is that normal/expected? 
> Personally, I would expect only the one I'd click to close?
> 
> https://github.com/dnbexperience/eufemia/assets/1359205/ff9314fe-a550-4ab7-9d6e-a6194f34fc0f
> 
> 
> I have not given this much thought(and not looked into the code of the example, etc), and it probably doesn't matter for the example which demos the "close all functionality", but I just thought of sharing this.

Nice catch! 
That does not look like the intended behaviour 😅

File: packages/dnb-design-system-portal/src/docs/uilib/components/accordion/demos.mdx [hunk 1]
```diff
@@ -67,3 +68,23 @@ Accordion can be disabled, though is not exactly defined what the use case is.
 <VisibleWhenVisualTest>
   <AccordionDescriptionExample />
 </VisibleWhenVisualTest>
+
+### Close All Accordions In A Group
+
+You can collapse all expanded accordions by sending a ref to the `closeAllHandleRef` prop and calling the `.current()` function on your ref.
+
+```jsx
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/components/accordion/demos.mdx:76]:
```suggestion
```tsx
```

File: packages/dnb-eufemia/src/components/accordion/Accordion.tsx [hunk 2]
```diff
@@ -228,6 +229,15 @@ function Accordion({
     }
   }, [context.flush_remembered_state, context.expanded_id])
 
+  // Add callback for closing all accordions inside a group
+  if (
+    !hasAddedCallbackRef.current &&
+    context?.collapseAccordionCallbacks
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/accordion/Accordion.tsx:235]:
```suggestion
    context?.collapseAccordionCallbacks &&
    !hasAddedCallbackRef.current
```
If the first does not match the second don't needs to run. When opposite, the second needs to run always.

File: packages/dnb-eufemia/src/components/accordion/AccordionDocs.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
+import { PropertiesTableProps } from '../../shared/types'
+
+export const accordionProperties: PropertiesTableProps = {
+  id: {
+    doc: 'a unique `id` that will be used on the button element. If you use `remember_state`, an id is required.',
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/accordion/AccordionDocs.ts:5]:
```suggestion
    doc: 'A unique `id` that will be used on the button element. If you use `remember_state`, an id is required.',
```

Good if we rule the docs the same and start capitalized on every single doc property.

File: packages/dnb-eufemia/src/components/accordion/AccordionDocs.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
+import { PropertiesTableProps } from '../../shared/types'
+
+export const accordionProperties: PropertiesTableProps = {
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/accordion/AccordionDocs.ts:3]:
```suggestion
export const AccordionProperties: PropertiesTableProps = {
```

For now, most of the typed docs run with a PascalCase naming, so its the same as the "Component" name.

`<PropertiesTable props={AccordionProviderGroupProperties} />` etc. needs to be adjusted too.

File: packages/dnb-eufemia/src/components/accordion/AccordionGroup.tsx [hunk 5]
```diff
@@ -58,7 +58,7 @@ const AccordionGroup = (props: AccordionGroupProps) => {
     expanded_id, // eslint-disable-line
     prerender, // eslint-disable-line
     prevent_rerender, // eslint-disable-line
-    single_container, // eslint-disable-line
+    single_container, // eslint-disable-linecloseAllHandleRef
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/accordion/AccordionGroup.tsx:61]:
```suggestion
    single_container, // eslint-disable-line
```

File: packages/dnb-eufemia/src/components/accordion/__tests__/Accordion.test.tsx [hunk 6]
```diff
@@ -277,6 +277,55 @@ describe('Accordion group component', () => {
...
+
+    const [first, second, third] = Array.from(
+      document.querySelectorAll('.dnb-accordion')
+    )
+
+    expect(first.className).toContain('--expanded')
+    expect(second.className).toContain('--expanded')
+    expect(third.className).toContain('--expanded')
+
+    act(() => closeAll.current())
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/accordion/__tests__/Accordion.test.tsx:323]:
```suggestion
    act(closeAll.current)
```

Up to you 😜

File: packages/dnb-eufemia/src/components/accordion/__tests__/Accordion.test.tsx [hunk 7]
```diff
@@ -277,6 +277,55 @@ describe('Accordion group component', () => {
...
+
+    expect(first.className).not.toContain('--expanded')
+    expect(second.className).not.toContain('--expanded')
+    expect(third.className).not.toContain('--expanded')
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/accordion/__tests__/Accordion.test.tsx:327]:
```suggestion
    expect(third).not.toHaveClass('--expanded')
```

Up to you.

File: packages/dnb-eufemia/src/components/accordion/__tests__/Accordion.test.tsx [hunk 8]
```diff
@@ -277,6 +277,55 @@ describe('Accordion group component', () => {
...
+      </Accordion.Group>
+    )
+
+    const [first, second, third] = Array.from(
+      document.querySelectorAll('.dnb-accordion')
+    )
+
+    expect(first.className).toContain('--expanded')
+    expect(second.className).toContain('--expanded')
+    expect(third.className).toContain('--expanded')
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/accordion/__tests__/Accordion.test.tsx:321]:
```suggestion
    expect(third).toHaveClass('--expanded')
```

Up to you.

**Review by tujoworker — COMMENTED** [review]
Since the term collapse is more commonly used and we also used in the the accordion table, should we maybe also call it for `collapseAllHandleRef`?

**Comment by joakbjerk** [comment]
> Since the term collapse is more commonly used and we also used in the the accordion table, should we maybe also call it for `collapseAllHandleRef`?

Can do 🙏

File: packages/dnb-eufemia/src/components/accordion/Accordion.tsx (hunk 2 above)
Comment (joakbjerk) [reply packages/dnb-eufemia/src/components/accordion/Accordion.tsx:235]: Great point 👍

File: packages/dnb-eufemia/src/components/accordion/AccordionDocs.ts (hunk 3 above)
Comment (joakbjerk) [reply packages/dnb-eufemia/src/components/accordion/AccordionDocs.ts:5]: Will fix the capitalization!

**Review by tujoworker — APPROVED** [review]
Nice work. Thank you ⭐

**Pull request merged**